homeassistant/components/permobil/config_flow.py

```diff
@@ -0,0 +1,198 @@
+"""Config flow for MyPermobil integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from mypermobil import MyPermobil, MyPermobilAPIException, MyPermobilClientException
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_CODE, CONF_EMAIL, CONF_REGION, CONF_TOKEN, CONF_TTL
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import APPLICATION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+GET_EMAIL_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.EMAIL)
+        ),
+    }
+)
+
+GET_TOKEN_SCHEMA = vol.Schema({vol.Required(CONF_CODE): cv.string})
+
+
+class InvalidAuth(exceptions.HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+async def validate_input(p_api: MyPermobil, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect."""
+    if email := data.get(CONF_EMAIL):
+        p_api.set_email(email)
+    if code := data.get(CONF_CODE):
+        p_api.set_code(code)
+    if token := data.get(CONF_TOKEN):
+        p_api.set_token(token)
+
+
+class PermobilConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Permobil config flow."""
+
+    VERSION = 1
+    p_api: MyPermobil = None
+    region_names: dict[str, str] = {"Failed to load regions": ""}
+    data: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Invoke when a user initiates a flow via the user interface."""
+        errors: dict[str, str] = {}
+        session = async_get_clientsession(self.hass)
+        self.p_api = MyPermobil(APPLICATION, session=session)
+
+        if user_input is not None:
+            user_input[CONF_EMAIL] = user_input[CONF_EMAIL]
+            try:
+                await validate_input(self.p_api, user_input)
+            except MyPermobilClientException as err:
+                _LOGGER.error("Error validating email: %s", err)
+                errors["base"] = "invalid_email"
+
+            self.data[CONF_EMAIL] = user_input[CONF_EMAIL]
```

Comment:
```suggestion
            self.data.update(user_input)
```

user_input contains only the email